src/views/docs-view/components/docs-root-landing/index.tsx

```diff
@@ -0,0 +1,55 @@
+import { Children, ReactChild } from 'react'
+import LandingHero from 'components/landing-hero'
+
+function getElementTextContent(child) {
+	let textContent = ''
+
+	Children.forEach(child.props.children, (child) => {
+		if (typeof child === 'string') {
+			textContent += child
+		}
+	})
+
+	return textContent
+}
+
+/**
+ * TEMPORARY
+ *
+ * A structured page view that wraps rendered MDX content, extracts the first rendered H1 and p tags and renders a
+ * landing hero in their place. Used to demonstrate structured page views pattern driven by frontmatter
+ */
+export default function DocsRootLanding({
+	children,
+}: {
+	children: ReactChild[]
+}) {
+	let heading
+	let subtitle
+
+	const childrenToRender = Children.map(children, (child) => {
+		if (typeof child === 'string' || typeof child === 'number') {
+			return child
+		}
+
+		if (!heading && child.props.originalType === 'h1') {
+			heading = {
+				title: getElementTextContent(child),
+				id: child.props.id,
+			}
+			return null
+		}
+		if (!subtitle && child.props.originalType === 'p') {
```

Comment:
It seems like there's potential for some strange edge cases here.

For example, if there's no `h1` in a document, which I think isn't guarded in other ways, we'd end up with an empty title element.

Or, if an `h1` is followed by a non-`p` element (and a `p` element appears later in the document), content would be re-ordered.

I wonder whether this might be a decent case for a `remark` plugin - it seems like we might be able to precisely target `h1 + p` at the start of a document on a given page with slightly more precision. We could replace it with a `dynamic` loaded custom MDX component.